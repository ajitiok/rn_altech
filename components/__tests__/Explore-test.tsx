import React from 'react';
import { render } from '@testing-library/react-native';
import TabTwoScreen from '../../app/(tabs)/explore';

describe('TabTwoScreen', () => {
  it('renders correctly', () => {
    const { getByText } = render(<TabTwoScreen />);
    expect(getByText('Explore')).toBeTruthy();
    expect(getByText('This app uses the following technologies:')).toBeTruthy();
  });

  it('contains the correct number of collapsible sections', () => {
    const { getAllByText } = render(<TabTwoScreen />);
    const collapsibleSections = getAllByText(/Technologies Used/i);
    expect(collapsibleSections.length).toBe(1);
  });
});