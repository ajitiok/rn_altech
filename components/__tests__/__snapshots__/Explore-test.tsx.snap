// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabTwoScreen renders correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      {
        "flex": 1,
      },
    ]
  }
>
  <RCTScrollView
    collapsable={false}
    forwardedRef={[Function]}
    jestAnimatedStyle={
      {
        "value": {},
      }
    }
    scrollEventThrottle={16}
  >
    <View>
      <View
        collapsable={false}
        jestAnimatedStyle={
          {
            "value": {
              "transform": [
                {
                  "translateY": 0,
                },
                {
                  "scale": 1,
                },
              ],
            },
          }
        }
        style={
          {
            "backgroundColor": "#D0D0D0",
            "height": 250,
            "overflow": "hidden",
            "transform": [
              {
                "translateY": 0,
              },
              {
                "scale": 1,
              },
            ],
          }
        }
      >
        <Text />
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "#fff",
            },
            {
              "flex": 1,
              "gap": 16,
              "overflow": "hidden",
              "padding": 32,
            },
          ]
        }
      >
        <View
          style={
            [
              {
                "backgroundColor": "#fff",
              },
              {
                "flexDirection": "row",
                "gap": 8,
              },
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "#11181C",
                },
                undefined,
                {
                  "fontSize": 32,
                  "fontWeight": "bold",
                  "lineHeight": 32,
                },
                undefined,
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            Explore
          </Text>
        </View>
        <Text
          style={
            [
              {
                "color": "#11181C",
              },
              {
                "fontSize": 16,
                "lineHeight": 24,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
            ]
          }
        >
          This app uses the following technologies:
        </Text>
        <View
          style={
            [
              {
                "backgroundColor": "#fff",
              },
              undefined,
            ]
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "gap": 6,
                "opacity": 1,
              }
            }
          >
            <Text />
            <Text
              style={
                [
                  {
                    "color": "#11181C",
                  },
                  undefined,
                  undefined,
                  {
                    "fontSize": 16,
                    "fontWeight": "600",
                    "lineHeight": 24,
                  },
                  undefined,
                  undefined,
                  undefined,
                ]
              }
            >
              Technologies Used
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
